### PR TITLE
Fixed #108 CoreOS osInfo

### LINKED ISSUE #108: CNI plugins show "index out of range" on CentOS 7.4 for kubernetes 1.8.9
**Is this a request for help?**:
Yes
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUES
---

**Which release version?**:
v1.0.3
---

**Which componenet (CNI/IPAM/CNM/CNS)**:
CNI
---

**Which Orchestrator and version (e.g. Kubernetes, Docker)**
Kubernetes v1.8.9, CentOS 7.4

**What happened**:

The CNI does not work after installation, the error from /var/log/messages asbelow:

Mar 31 10:24:17 localhost kubelet: 2018/03/31 10:24:17 File not exist /var/run/AzureCNITelemetry.json
Mar 31 10:24:17 localhost kubelet: 2018/03/31 10:24:17 GetReport state file didn't exist. Setting flag to true
Mar 31 10:24:17 localhost kubelet: panic: runtime error: index out of range
Mar 31 10:24:17 localhost kubelet: goroutine 1 [running]:
Mar 31 10:24:17 localhost kubelet: github.com/Azure/azure-container-networking/telemetry.(*Report).GetOSDetails(0xc4200d0640)
Mar 31 10:24:17 localhost kubelet: /go/src/github.com/Azure/azure-container-networking/telemetry/telemetry_linux.go:112 +0x55c
Mar 31 10:24:17 localhost kubelet: github.com/Azure/azure-container-networking/telemetry.(*ReportManager).GetReport(0xc420012340, 0x6ea639, 0x3, 0x70d874, 0x6)
Mar 31 10:24:17 localhost kubelet: /go/src/github.com/Azure/azure-container-networking/telemetry/telemetry.go:138 +0xa3
Mar 31 10:24:17 localhost kubelet: main.main()
Mar 31 10:24:17 localhost kubelet: /go/src/github.com/Azure/azure-container-networking/cni/network/plugin/main.go:65 +0x6e6
Mar 31 10:24:17 localhost kubelet: E0331 10:24:17.369675  103320 cni.go:319] Error deleting network: netplugin failed but error parsing its diagnostic message "": unexpected end of JSON input
Mar 31 10:24:17 localhost dockerd-current: time="2018-03-31T10:24:17.370528075Z" level=error msg="Handler for POST /v1.26/containers/f2846470d4734c61bee35b9e76bcff3ca9d704ff0b7320398ebe3da2a2a26163/stop returned error: Container f2846470d4734c61bee35b9e76bcff3ca9d704ff0b7320398ebe3da2a2a26163 is already stopped"
Mar 31 10:24:17 localhost kubelet: E0331 10:24:17.371366  103320 remote_runtime.go:115] StopPodSandbox "f2846470d4734c61bee35b9e76bcff3ca9d704ff0b7320398ebe3da2a2a26163" from runtime service failed: rpc error: code = Unknown desc = NetworkPlugin cni failed to teardown pod "kube-dns-545bc4bfd4-pzc4c_kube-system" network: netplugin failed but error parsing its diagnostic message "": unexpected end of JSON input
Mar 31 10:24:17 localhost kubelet: E0331 10:24:17.371439  103320 kuberuntime_manager.go:784] Failed to stop sandbox {"docker" "f2846470d4734c61bee35b9e76bcff3ca9d704ff0b7320398ebe3da2a2a26163"}
Mar 31 10:24:17 localhost kubelet: E0331 10:24:17.371537  103320 kuberuntime_manager.go:584] killPodWithSyncResult failed: failed to "KillPodSandbox" for "6b0cd455-34c6-11e8-a328-000d3a1ab23a" with KillPodSandboxError: "rpc error: code = Unknown desc = NetworkPlugin cni failed to teardown pod \"kube-dns-545bc4bfd4-pzc4c_kube-system\" network: netplugin failed but error parsing its diagnostic message \"\": unexpected end of JSON input"
Mar 31 10:24:17 localhost kubelet: E0331 10:24:17.371587  103320 pod_workers.go:182] Error syncing pod 6b0cd455-34c6-11e8-a328-000d3a1ab23a ("kube-dns-545bc4bfd4-pzc4c_kube-system(6b0cd455-34c6-11e8-a328-000d3a1ab23a)"), skipping: failed to "KillPodSandbox" for "6b0cd455-34c6-11e8-a328-000d3a1ab23a" with KillPodSandboxError: "rpc error: code = Unknown desc = NetworkPlugin cni failed to teardown pod \"kube-dns-545bc4bfd4-pzc4c_kube-system\" network: netplugin failed but error parsing its diagnostic message \"\": unexpected end of JSON input"

**What you expected to happen**:
The CNI for kubernetes should work properly on CentOS 7.X

**How to reproduce it** (as minimally and precisely as possible):
1. Find an CentOS 7.4 VM
2. Install kubeadm, install kubernetes 1.8.9
3. Install Azure CNI plugins for kubernetest on CentOS
3. Check /var/log/messages, kubectl describe nodes

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by IvanovOleg**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #108 

**Special notes for your reviewer**:
Could you please at least build it and upload artifacts somewhere? Because this problem blocks me and I can't build it myself because of https://github.com/Azure/azure-container-networking/issues/148
**Release note**:

```release-note: 
Use /etc/os-release instead of /etc/issue as a source for the osInfo because /etc/issue contains completely different lines depending on the linux distributive and fails on the CoreOS
```

### CODE REVIEW

**Comment by IvanovOleg** [comment]
Could you please at least build it and upload artifacts somewhere? Because this problem blocks me and I can't build it myself because of https://github.com/Azure/azure-container-networking/issues/148

**Comment by IvanovOleg** [comment]
Another changes related to the build issues #148 went to my fork, sorry. I can prepare another PR if you want. Let me know.

File: telemetry/telemetry_linux.go [hunk 1]
```diff
@@ -88,14 +88,19 @@ func (report *Report) GetSystemDetails() {
...
-		report.OSDetails.ErrorMessage = "reading /etc/issue failed with" + err.Error()
+		report.OSDetails.ErrorMessage = "reading /etc/os-release failed with" + err.Error()
 		return
 	}
 
-	osInfoArr := strings.Split(linesArr[0], " ")
+	osInfoArr := make(map[string]string)
+	
+	for i := range linesArr {
+                s := strings.Split(linesArr[i], "=")
```
[the remark this answers is not in the document]
Comment (IvanovOleg) [reply telemetry/telemetry_linux.go]: It works, but you cat check it yourself if you want. I've checked this split by isolating the function and printing the result, it worked. Then I've build the artifact from my fork and use it with my kubernetes 1.10.3. That issue disappeared.

**Comment by IvanovOleg** [comment]
I am going to prepare two PRs instead of this one.

**Comment by IvanovOleg** [comment]
https://github.com/Azure/azure-container-networking/pull/154

**Comment by tamilmani1989** [comment]
@IvanovOleg closing this PR as you opened a new PR #154

**Pull request closed without merging**